pcre.go

```diff
@@ -602,6 +602,35 @@ func (re Regexp) ReplaceAllString(in, repl string, flags int) string {
 	return string(re.ReplaceAll([]byte(in), []byte(repl), flags))
 }
 
+// Match holds details about a single successful regex match.
+type Match struct {
+	Finding string // Text that was found.
+	Loc     []int  // Index bounds for location of finding.
+}
+
+// FindAll finds all instances that match the regex.
+func (re Regexp) FindAll(subject string, flags int) []Match {
+	matches := make([]Match, 0)
+	m := re.MatcherString(subject, flags)
+	offset := 0
```

Comment:
Instead of maintaining an `offset` variable and doing index math, I think it would be cleaner to just use a slice variable that you update on each iteration. Then you should be able to use the indices in `m.ovector` directly on that slice, instead of indexing into the original string.

Reply:
Need the Location to return absolute indices for proximity based regexes. Not keeping a separate slice for indexing makes it more consistent between extracting the Finding and the Location of the finding.

Reply:
Ahh got it, ok